Modules/_sqlite/row.c

```diff
@@ -204,6 +204,84 @@ static PyObject* pysqlite_row_richcompare(pysqlite_Row *self, PyObject *_other,
     Py_RETURN_NOTIMPLEMENTED;
 }
 
+static PyObject *
+pysqlite_row_repr(pysqlite_Row *self)
+{
+    _PyUnicodeWriter writer;
+    int result;
+    Py_ssize_t nitems, i;
+    PyObject *start;
+    PyObject *name;
+    PyObject *value;
+
+    _PyUnicodeWriter_Init(&writer);
+
+    start = PyUnicode_FromFormat("<%s object; {", Py_TYPE(self)->tp_name);
+    if (start == NULL) {
+        goto error;
+    }
+    result = _PyUnicodeWriter_WriteStr(&writer, start);
+    Py_DECREF(start);
+    if (result < 0) {
+        goto error;
+    }
+
+    nitems = PyTuple_Size(self->description);
+    for (i = 0; i < nitems; i++) {
+        if (i > 0) {
+            if (_PyUnicodeWriter_WriteASCIIString(&writer, ", ", 2) < 0) {
+                goto error;
+            }
+        }
+
+        name = PyTuple_GET_ITEM(self->description, i);
+        if (name == NULL) {
+            goto error;
```

Comment:
You don't have to do a `PY_DECREF(name)` before `goto error;`?

Reply:
@eamanu in this case, name will be a `NULL` pointer.

Reply:
Also note that `PyTuple_GET_ITEM()` returns borrowed reference (which means it doesn't incref it), so I don't have to decref even on success.